File: sdks/php/tests/Api/DIDCommApiTest.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Tests\Api;

use Atap\Sdk\Model\DIDCommInbox;
use Atap\Sdk\Tests\TestCase;

class DIDCommApiTest extends TestCase
{
    public function testSend(): void
    {
        $client = $this->createUnauthClient([
            $this->jsonResponse(202, [
                'id' => 'msg_123',
                'status' => 'queued',
            ]),
        ]);

        $result = $client->didcomm()->send('{"protected":"...","ciphertext":"..."}');
        $this->assertSame('msg_123', $result['id']);
        $this->assertSame('queued', $result['status']);
    }

    public function testInbox(): void
    {
        $client = $this->createAuthedClient([
            $this->jsonResponse(200, [
                'messages' => [
                    [
                        'id' => 'msg_1',
                        'sender_did' => 'did:web:sender',
                        'message_type' => 'text',
                        'payload' => 'Hello',
                        'created_at' => '2024-01-01T00:00:00Z',
                    ],
                    [
                        'id' => 'msg_2',
                        'sender_did' => 'did:web:sender2',
                        'message_type' => 'json',
                        'payload' => '{"key":"val"}',
                        'created_at' => '2024-01-01T00:01:00Z',
                    ],
                ],
                'count' => 2,
            ]),
        ]);

        $inbox = $client->didcomm()->inbox();
        $this->assertInstanceOf(DIDCommInbox::class, $inbox);
        $this->assertCount(2, $inbox->messages);
        $this->assertSame(2, $inbox->count);
        $this->assertSame('msg_1', $inbox->messages[0]->id);
        $this->assertSame('did:web:sender', $inbox->messages[0]->senderDid);
    }

    public function testInboxEmpty(): void
    {
        $client = $this->createAuthedClient([
            $this->jsonResponse(200, [
                'messages' => [],
                'count' => 0,
            ]),
        ]);

        $inbox = $client->didcomm()->inbox();
        $this->assertCount(0, $inbox->messages);
        $this->assertSame(0, $inbox->count);
    }

    public function testInboxWithLimit(): void
    {
        $client = $this->createAuthedClient([
            $this->jsonResponse(200, [
                'messages' => [
                    ['id' => 'msg_1', 'sender_did' => 'a', 'message_type' => 'x', 'payload' => '', 'created_at' => ''],
                ],
                'count' => 1,
            ]),
        ]);

        $inbox = $client->didcomm()->inbox(10);
        $this->assertSame(1, $inbox->count);
    }
}

File: sdks/java/src/test/java/dev/atap/sdk/api/DIDCommApiTest.java
package dev.atap.sdk.api;

import com.github.tomakehurst.wiremock.WireMockServer;
import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
import dev.atap.sdk.ATAPClient;
import dev.atap.sdk.TestFixtures;
import dev.atap.sdk.model.DIDCommInbox;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static com.github.tomakehurst.wiremock.client.WireMock.*;
import static org.junit.jupiter.api.Assertions.*;

class DIDCommApiTest {

    private WireMockServer wireMock;
    private ATAPClient client;

    @BeforeEach
    void setUp() {
        wireMock = new WireMockServer(WireMockConfiguration.wireMockConfig().dynamicPort());
        wireMock.start();

        wireMock.stubFor(post(urlEqualTo("/v1/oauth/token"))
                .willReturn(aResponse()
                        .withStatus(200)
                        .withHeader("Content-Type", "application/json")
                        .withBody(TestFixtures.tokenResponseJson())));

        client = ATAPClient.builder()
                .baseUrl("http://localhost:" + wireMock.port())
                .did(TestFixtures.TEST_DID)
                .keyPair(TestFixtures.getTestKeyPair())
                .clientSecret(TestFixtures.TEST_CLIENT_SECRET)
                .platformDomain(TestFixtures.TEST_PLATFORM_DOMAIN)
                .build();
    }

    @AfterEach
    void tearDown() {
        wireMock.stop();
        client.close();
    }

    @Test
    void testSend() {
        wireMock.stubFor(post(urlEqualTo("/v1/didcomm"))
                .willReturn(aResponse()
                        .withStatus(200)
                        .withHeader("Content-Type", "application/json")
                        .withBody("{\"id\":\"msg_1\",\"status\":\"queued\"}")));

        Map<String, Object> result = client.didcomm.send("jwe_bytes".getBytes());
        assertNotNull(result);
        assertEquals("queued", result.get("status"));
    }

    @Test
    void testInbox() {
        wireMock.stubFor(get(urlPathEqualTo("/v1/didcomm/inbox"))
                .willReturn(aResponse()
                        .withStatus(200)
                        .withHeader("Content-Type", "application/json")
                        .withBody(TestFixtures.didcommInboxJson())));

        DIDCommInbox inbox = client.didcomm.inbox();
        assertNotNull(inbox);
        assertEquals(1, inbox.getCount());
        assertEquals(1, inbox.getMessages().size());
        assertEquals("msg_1", inbox.getMessages().get(0).getId());
        assertEquals("did:web:example:agent:sender", inbox.getMessages().get(0).getSenderDid());
        assertEquals("notification", inbox.getMessages().get(0).getMessageType());
    }

    @Test
    void testInboxWithLimit() {
        wireMock.stubFor(get(urlPathEqualTo("/v1/didcomm/inbox"))
                .withQueryParam("limit", equalTo("10"))
                .willReturn(aResponse()
                        .withStatus(200)
                        .withHeader("Content-Type", "application/json")
                        .withBody("{\"messages\":[],\"count\":0}")));

        DIDCommInbox inbox = client.didcomm.inbox(10);
        assertNotNull(inbox);
        assertEquals(0, inbox.getCount());
        assertTrue(inbox.getMessages().isEmpty());
    }

    @Test
    void testInboxLimitCappedAt100() {
        wireMock.stubFor(get(urlPathEqualTo("/v1/didcomm/inbox"))
                .withQueryParam("limit", equalTo("100"))
                .willReturn(aResponse()
                        .withStatus(200)
                        .withHeader("Content-Type", "application/json")
                        .withBody("{\"messages\":[],\"count\":0}")));

        DIDCommInbox inbox = client.didcomm.inbox(200);
        assertNotNull(inbox);
    }
}
